samples/algorithms/sudoku-grover/Sudoku.qs

```diff
@@ -0,0 +1,151 @@
+    /// # Output
+    /// A Tuple with Result and the array of numbers for each empty square
+    /// Look at the README.md sample output to see examples of what this is for different sample puzzles
+    /// # Remarks
+    /// The inputs and outputs for the following 4x4 puzzle are
+    ///    -----------------
+    ///    |   | 1 | 2 | 3 |         <--- empty square #0
+    ///    -----------------
+    ///    | 2 |   | 0 | 1 |         <--- empty square #1
+    ///    -----------------
+    ///    | 1 | 2 | 3 | 0 |
+    ///    -----------------
+    ///    | 3 |   | 1 | 2 |         <--- empty square #2
+    ///    -----------------
+    ///    emptySquareEdges = [(1, 0),(2, 1)]    
+    ///         empty square #0 can not have the same color/number as empty call #1 because they are diagonal
+    ///         empty square #1 and #2 can not have the same color/number because they are in the same column
+    ///    startingNumberConstraints = [(0, 2),(0, 1),(0, 3),(1, 1),(1, 2),(1, 0),(2, 1),(2, 2),(2, 3)]
+    ///         empty square #0 can not have values 2,1,3 because those are in the same row/column/2x2grid
+    ///         empty square #1 can not have values 1,2,0 because those are in the same row/column/2x2grid
+    ///    Results = [0,3,0] i.e. Empty Square #0 = 0, Empty Square #1 = 3, Empty Square #2 = 0
+    operation SolvePuzzle(V : Int, size: Int, emptySquareEdges: (Int, Int)[], startingNumberConstraints: (Int, Int)[]) : (Bool, Int[]) {
+        mutable bitsPerColor = 2; // if size == 4x4 grid
+        if (size == 9)
+        {
+            set bitsPerColor = 4; // if size == 9x9 grid
+        }
+        let numIterations = NIterations(bitsPerColor * V);
+        Message($"Running Quantum test with #Vertex = {V}");
+        Message($"   Bits Per Color = {bitsPerColor}");
+        Message($"   emptySquareEdges = {emptySquareEdges}");
+        Message($"   startingNumberConstraints = {startingNumberConstraints}");
+        Message($"   estimated #iterations needed = {numIterations}");
+        Message($"   size of Sudoku grid = {size}x{size}");
+        mutable coloring = new Int[0];
+        if (size == 4) {
+            set coloring = GroversAlgorithm(V, 2, numIterations*2, VertexColoringOracle(V, 2, emptySquareEdges, startingNumberConstraints, _, _));
+        }
+        elif (size == 9) {
+            set coloring = GroversAlgorithm(V, 4, numIterations*2, VertexColoringOracle4Bit9Color(V, emptySquareEdges, startingNumberConstraints, _, _));
+        }
+
+        Message($"Got sudoku solution: {coloring}");
+        if (IsSudokuSolutionValid(V, size, emptySquareEdges, startingNumberConstraints, coloring)) {
+           Message($"Got valid sudoku solution: {coloring}");
+           return (true, coloring);
+        } else {
+           Message($"Got invalid sudoku solution: {coloring}");
+           return (false, coloring);
+        }
+    }
+
+    /// # Summary
+    /// Estimate the number of interations required for solution
```

Comment:
I think it would be useful to also add inputs and outputs to the docstring here.